pkg/build/controller/controller.go

```diff
@@ -216,15 +216,24 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	case kapi.PodRunning:
 		// The pod's still running
 		nextStatus = buildapi.BuildPhaseRunning
-	case kapi.PodSucceeded, kapi.PodFailed:
+	case kapi.PodSucceeded:
 		// Check the exit codes of all the containers in the pod
 		nextStatus = buildapi.BuildPhaseComplete
-		for _, info := range pod.Status.ContainerStatuses {
-			if info.State.Terminated != nil && info.State.Terminated.ExitCode != 0 {
-				nextStatus = buildapi.BuildPhaseFailed
-				break
+		if len(pod.Status.ContainerStatuses) == 0 {
```

Comment:
@bparees but if the pod indeed failed then why do you have this check under the succeeded pod case?